src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -637,6 +641,49 @@ public void AssemblyResolve_RequestingAssembly()
             }).Dispose();
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Does not support AssemblyLoadContext")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Does not support AssemblyLoadContext")]
+        public void AssemblyResolve_FirstChanceException()
+        {
+            RemoteExecutor.Invoke(() => {
+                Assembly assembly = typeof(AppDomainTests).Assembly;
+
+                bool firstChanceExceptionThrown = false;
+
+                EventHandler<System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs> firstChanceHandler = (source, args) =>
+                {
+                    firstChanceExceptionThrown = true;
+                };
+
+                AppDomain.CurrentDomain.FirstChanceException += firstChanceHandler;
+
+                ResolveEventHandler assemblyResolveHandler = (sender, e) =>
+                {
+                    Assert.Equal(assembly, e.RequestingAssembly);
+                    Assert.Equal(false, firstChanceExceptionThrown);
+                    return null;
+                };
+
+                AppDomain.CurrentDomain.AssemblyResolve += assemblyResolveHandler;
+
+                Func<System.Runtime.Loader.AssemblyLoadContext, AssemblyName, Assembly> resolvingHandler = (context, name) =>
+                {
+                    return null;
+                };
+
+                System.Runtime.Loader.AssemblyLoadContext.Default.Resolving += resolvingHandler;
+
+                assembly.GetType("System.Tests.AGenericClass`1[[Bogus, BogusAssembly]]", false);
+                Assert.Equal(false, firstChanceExceptionThrown);
+
+                Assert.Throws<FileNotFoundException>(() => assembly.GetType("System.Tests.AGenericClass`1[[Bogus, AnotherBogusAssembly]]", true));
+                Assert.Equal(true, firstChanceExceptionThrown);
```

Comment:
Nit: `Assert.True(firstChanceExceptionThrown)`

Is there anything we can validate beyond this?  For example, rather than just setting a bool to true, can we store the actual Exception instance that was thrown, and then Assert.Same that here against the one returned from GetType?